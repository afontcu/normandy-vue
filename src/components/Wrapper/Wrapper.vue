<template>
  <div class="o-wrapper" :class="[ classes ]">
    <slot></slot>
  </div>
</template>

<script>
import VueTypes from 'vue-types'
import { SIZE_VALUES } from '@/constraints'

export default {
  name: 'Wrapper',
  props: {
    size: VueTypes.oneOf(SIZE_VALUES)
  },
  data () {
    return {}
  },
  computed: {
    classes () {
      return {
        [`o-wrapper--${this.size}`]: this.size
      }
    }
  }
}
</script>
