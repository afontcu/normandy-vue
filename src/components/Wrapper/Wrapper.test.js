import Wrapper from './Wrapper'
import { mount } from 'avoriaz'
import { compileToFunctions } from 'vue-template-compiler'

describe('Wrapper.vue', () => {
  beforeAll(() => {
    console.error = jest.fn(error => { throw new Error(error) })
    console.warn = jest.fn(warn => { throw new Error(warn) })
  })

  it('renders correct nodes when required props are provided', () => {
    const wrapper = mount(Wrapper)
    expect(wrapper.hasClass('o-wrapper')).toBe(true)
  })

  it('renders slot content when defined', () => {
    const compiled = compileToFunctions('<div class="dummy"></div>')
    const wrapper = mount(Wrapper, { slots: { default: compiled } })
    expect(wrapper.first('.dummy')).toBeTruthy()
  })

  it('adds size class when size prop is provided', () => {
    const values = ['flush', 'tiny', 'small', 'large', 'huge']

    values.forEach((size) => {
      const wrapper = mount(Wrapper)
      wrapper.setProps({ size })
      expect(wrapper.hasClass(`o-wrapper--${size}`)).toBe(true)
    })
  })

  it('changes size classes when prop changes', () => {
    const wrapper = mount(Wrapper)

    wrapper.setProps({ size: 'large' })
    expect(wrapper.hasClass('o-wrapper--large')).toBe(true)
    expect(wrapper.hasClass('o-wrapper--small')).toBe(false)

    wrapper.setProps({ size: 'small' })
    expect(wrapper.hasClass('o-wrapper--small')).toBe(true)
    expect(wrapper.hasClass('o-wrapper--large')).toBe(false)
  })

  it('breaks with invalid size prop value', () => {
    const size = 'invalidProp'
    expect(() => {
      mount(Wrapper, { propsData: { size } })
    }).toThrow()
  })

  it('renders an snapshot', () => {
    const wrapper = mount(Wrapper)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
