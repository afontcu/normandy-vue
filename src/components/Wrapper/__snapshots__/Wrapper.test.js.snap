// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper.vue renders an snapshot 1`] = `

<div class="o-wrapper">
</div>

`;
